src/main/java/us/abstracta/wiresham/FlowFactory.java

```diff
@@ -0,0 +1,187 @@
+        flows.get(0).setPort(port);
+      }
+      return flows;
+    } catch (IOException e) {
+      LOG.error("There was an error trying to read the wireshark flow", e);
+      System.err.println(e.getMessage());
+      return null;
+    }
+  }
+
+  private List<Flow> buildFromYamlFile() throws FileNotFoundException {
+    List<PacketStep> packets = new Yaml(buildYamlConstructor())
+        .load(new FileInputStream(configFile));
+    return dissectStepsIntoFlow(packets);
+  }
+
+  private static Constructor buildYamlConstructor() {
+    Constructor constructor = new Constructor();
+    YAML_TAGS
+        .forEach((tag, clazz) -> constructor.addTypeDescription(new TypeDescription(clazz, tag)));
+    return constructor;
+  }
+
+  public List<Flow> dissectStepsIntoFlow(List<PacketStep> packetSteps) {
+    List<List<PacketStep>> dissectedFlow = new ArrayList<>();
+    List<PacketStep> currentFlowPort = new ArrayList<>();
+    for (PacketStep step : packetSteps) {
+      if ((step instanceof SendPacketStep) && ((SendPacketStep) step).getPort() != null) {
+        if (!currentFlowPort.isEmpty()) {
+          dissectedFlow.add(currentFlowPort);
+        }
+        currentFlowPort = new ArrayList<>();
+      }
+      currentFlowPort.add(step);
+    }
+    dissectedFlow.add(currentFlowPort);
+    return buildFlowFromDissectedPackets(dissectedFlow);
+  }
+
+  private Flow buildWiresharkFlow() throws IOException {
+    if (configFile.getName().toLowerCase().endsWith(".json")) {
+      return Flow.fromWiresharkJsonDump(configFile, serverAddress);
+    } else {
+      return Flow.fromPcap(configFile, serverAddress, pcapFilter);
+    }
+  }
+
+  public void saveYml(File ymlFile) throws IOException {
+    new Yaml(buildYamlRepresenter())
+        .dump(getFlows(), new FileWriter(ymlFile));
+  }
+
+  private static Representer buildYamlRepresenter() {
+    Representer representer = new Representer() {
+      @Override
+      protected NodeTuple representJavaBeanProperty(Object javaBean, Property property,
+          Object propertyValue, Tag customTag) {
+        if (property.getType() == long.class && (long) propertyValue == 0) {
+          return null;
+        } else if (property.getName().equals("port") && (int) propertyValue == 0) {
+          return null;
+        } else {
+          return super.representJavaBeanProperty(javaBean, property, propertyValue, customTag);
+        }
+      }
+    };
+
+    YAML_TAGS.forEach((tag, clazz) -> representer.addClassTag(clazz, new Tag(tag)));
+    return representer;
+  }
+
+  public List<Flow> getReversedFlows() {
+    List<Flow> flows = getFlows();
+    List<List<PacketStep>> dissectedReversedPackets = new ArrayList<>();
+    for (Flow flow : flows) {
+      List<PacketStep> reversedPackets = flow.getSteps().stream()
+          .map(s -> s instanceof SendPacketStep
+              ? new ReceivePacketStep(s.data.toString())
+              : new SendPacketStep(s.data.toString(), 0))
```

Comment:
I think here goes the discussion we had on ports and client/server sockets, and my previous comment on client sockets port. Probably need to adapt this logic